test/hocon_tconf_tests.erl

```diff
@@ -720,9 +720,33 @@ required_test() ->
         #{<<"f2">> => "string", <<"f3">> => 0},
         hocon_tconf:check_plain(ScRequired, #{<<"f2">> => <<"string">>}, #{required => true})
     ),
-
     ok.
 
+deprection_test() ->
+    Sc = #{
+        roots => [
+            {f1, hoconsc:mk(integer())},
+            {f2, hoconsc:mk(string())},
+            {f3, hoconsc:mk(integer(), #{deprecated => true})}
+        ]
+    },
+    ?assertEqual(
+        #{<<"f2">> => "string"},
+        hocon_tconf:check_plain(
+            Sc,
+            #{<<"f2">> => <<"string">>},
+            #{required => false}
+        )
+    ),
+    ?assertEqual(
+        #{<<"f2">> => "string"},
+        hocon_tconf:check_plain(
+            Sc,
+            #{<<"f2">> => <<"string">>, <<"f3">> => 4444},
```

Comment:
Suggestion: maybe test that we can even change the type of the values and that won't be validated?